src/game/client/water/w3dwater.cpp

```diff
@@ -639,11 +641,11 @@ int WaterRenderObjClass::Init(float water_level, float dx, float dy, SceneClass
     m_shaderClass = g_zFillAlphaShader;
     m_shaderClass.Set_Cull_Mode(ShaderClass::CULL_MODE_DISABLE);
     m_alphaClippingTexture = W3DAssetManager::Get_Instance()->Get_Texture("TSMoonLarg.tga");
-    m_skyBox = ((GameAssetManager *)W3DAssetManager::Get_Instance())
+    m_skyBox = static_cast<GameAssetManager *>(W3DAssetManager::Get_Instance())

```

Comment:
Is there not a method that gets the specific type?

Reply:
I don't understand.

Reply:
Hmm no. `static W3DAssetManager *&s_theInstance;` is protected class member.